contracts/AstarBase.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./DappsStaking.sol";
+import "./SR25519.sol";
+
+/// @author The Astar Network Team
+/// @title Astarbase. Mapping of Stakers ss58 <> H160
+contract AstarBase is Ownable {
+    using Counters for Counters.Counter;
+    Counters.Counter public registeredCnt;
+    bytes PREFIX= hex"3c42797465733e";
+    bytes POSTFIX= hex"3c2f42797465733e";
+    bool public paused;
+    mapping(address => bytes32) public addressMap;
+    DappsStaking public constant DAPPS_STAKING = DappsStaking(0x0000000000000000000000000000000000005001);
+    SR25519 public constant SR25519Contract = SR25519(0x0000000000000000000000000000000000005002);
+
+    constructor() {
+        paused = false;
+    }
+
+    /// @notice Register senders' address with corresponding SS58 address and store to mapping
+    /// @param polkadotPublicKey, SS58 address used for staking
+    /// @param signedMsg, signed message
+    /// @param message, message string that was signed
+    function register(bytes32 polkadotPublicKey, bytes calldata signedMsg, string calldata message) external {
+        require(!paused, "The contract is paused");
+        require(polkadotPublicKey != 0, "Can't register polkadotPublicKey with 0");
```

Comment:
I think the var name `polkadotPublicKey` isn't good since Polkadot is a network name, not a standard. Also, the term public key is technically incorrect since sr25519 accounts will not expose the public key. What about using `substrateAccountId` or just `accountId`?